src/main/java/io/dropwizard/kafka/KafkaAdminClientBundle.java

```diff
@@ -0,0 +1,59 @@
+package io.dropwizard.kafka;
+
+import brave.Tracing;
+import io.dropwizard.Configuration;
+import io.dropwizard.ConfiguredBundle;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class KafkaAdminClientBundle<T extends Configuration> implements ConfiguredBundle<T> {
+    private final Collection<NewTopic> topics;
+    private final Map<String, Object> configOverrides;
+
+    @Nullable
+    private AdminClient adminClient;
```

Comment:
Why can this be nullable? In the getter you have requireNonNull ?

Reply:
Before the `run` method is called, it will be `null`. The requireNonNull is to prevent people from doing the wrong thing.